request_handler.h

```diff
@@ -14,12 +15,47 @@ class RequestHandler {
 		ERROR = 1
 	};
 
-	// requestData is the full http request
-	// request is the parsed header
+	// Initializes the handler. Returns a response code indicating success or
+	// failure condition.
+	// uri_prefix is the value in the config file that this handler will run for.
+	// config is the contents of the child block for this handler ONLY.
+	virtual Status Init(const std::string& uri_prefix, const NginxConfig& config) = 0;
+
+	// Creation Magic by name of handler as specified in config
+	static std::shared_ptr<RequestHandler> CreateByName(const std::string& type);
```

Comment:
Why shared_ptr? It can probably be a raw pointer. If you want it can be a unique pointer.